src/common/libutil/hola.h

```diff
@@ -0,0 +1,53 @@
+/************************************************************\
+ * Copyright 2022 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _UTIL_HOLA_H
+#define _UTIL_HOLA_H
+
+#include "src/common/libczmqcontainers/czmq_containers.h"
+
+enum {
+    HOLA_AUTOCREATE = 1,    // create list on first addition
+    HOLA_AUTODESTROY = 2,   // destroy list on last removal
```

Comment:
some comments would be good here to indicate they make it so user doesn't have to call `hola_hash_add()` or `hola_hash_delete()`.  I was initially confused what their purpose was, b/c I thought this behavior would be the default behavior.

Reply:
I was thinking there may be a use case for the class where the lists are only created intentionally and it would be desirable to fail an add to an unknown list.  Then of course when the list is emptied you wouldn't necessarily want it to be deleted.  Also, you may not want to delete empty lists if you expect a limited set of lists to transition back and forth from zero items and nonzero items, for performance.

I'll add a bit more doc.